src/regthreesixty.js

```diff
@@ -182,49 +205,75 @@ angular.module('reg.threesixty', [])
         element.on('touchstart mousedown', mousedown);
 
         function mousedown (event) {
-          event.preventDefault();
           pointerStartPosX = getPointerEvent(event).pageX;
+          pointerStartPosY = getPointerEvent(event).pageY;
           dragging = true;
 
-          $document.on('touchmove mousemove', mousemove);
-          $document.on('touchend mouseup', mouseup);
+          element.on('touchmove mousemove', mousemove);
+          element.on('touchend mouseup', mouseup);
         }
 
         function trackPointer(event){
-          if (ready && dragging) {
+          if (ready && dragging && !scrolling) {
+
+            var pointerEvent = getPointerEvent(event);
 
-            pointerEndPosX = getPointerEvent(event).pageX;
+            pointerEndPosX = pointerEvent.pageX;
+            pointerEndPosY = pointerEvent.pageY;
 
             if(monitorStartTime < new Date().getTime() - monitorInt) {
               var frameDiff = 0,
                 direction = scope.reverse? -1 : 1 ;
 
               pointerDistance = pointerEndPosX - pointerStartPosX;
+              var xDistanceAbs = Math.abs(pointerDistance);
+              var pointerDistanceY = Math.abs(pointerEndPosY - pointerStartPosY);
 
-              if(pointerDistance > 0){
-                frameDiff = Math.ceil((totalFrames - 1) * speedMultiplier * (pointerDistance / element[0].clientWidth));
-              }else{
-                frameDiff = Math.floor((totalFrames - 1) * speedMultiplier * (pointerDistance / element[0].clientWidth));
-              }
+              if (((!initialDrag && xDistanceAbs > 1) || (initialDrag && xDistanceAbs > 5)) &&
+                  (pointerDistanceY * 3) < xDistanceAbs) {
 
-              endFrame = currentFrame + (direction * frameDiff);
+                  if (initialDrag) {
+                    initialDrag = false;
+                    body.style.top = -Math.abs(scrollY) + 'px';
+                    bodyClasses.add('no-scroll');
+                  }
+
+                  if (pointerDistance > 0){
+                      frameDiff = Math.ceil((totalFrames - 1) * speedMultiplier * (pointerDistance / element[0].clientWidth));
```

Comment:
This and the argument of the `Math.floor` call on line 244 are identical. It would therefore be nicer to put it into a variable and use it here and on line 244.